generate.py

```diff
@@ -179,6 +185,15 @@ def main():
 
         # Run the WaveNet to predict the next sample.
         prediction = sess.run(outputs, feed_dict={samples: window})[0]
+
+        # Scale sample distribution using temperature, if applicable.
+        if (args.temp != 1.0 and args.temp > 0):
+            np.seterr(divide='ignore')
+            prediction = np.log(prediction) / args.temp
+            prediction[np.isneginf(prediction)] = 0
+            prediction = np.exp(prediction) / np.sum(np.exp(prediction))
```

Comment:
I think this line is better implemented as something like:
```
prediction = prediction - scipy.misc.logsumexp(prediction)
prediction = np.exp(prediction)
```
By operating in the log domain as much as possible we avoid divisions which can lead to instability when we divide by something very close to zero. Especially if we ever want to run using float16.